code/player/Inventory.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Linq;
+using Sandbox;
+
+using TTTReborn.Weapons;
+
+namespace TTTReborn.Player
+{
+    partial class Inventory : BaseInventory
+    {
+        public Inventory(TTTPlayer player) : base(player)
+        {
+
+        }
+
+        public override bool Add(Entity entity, bool makeActive = false)
+        {
+            TTTPlayer player = Owner as TTTPlayer;
+
+            if (IsCarryingType(entity.GetType()))
+            {
+                return false;
+            }
+
+            if (entity is Weapon weapon)
+            {
+                Sound.FromWorld("dm.pickup_weapon", entity.Position);
+            }
+
+            return base.Add(entity, makeActive);
+        }
+
+        public bool IsCarryingType(Type t)
+        {
+            return List.Any(x => x.GetType() == t);
+        }
+
+        int GetNextIndex(int index, int maxIndex)
+        {
+            return NormalizeSlotIndex(index + 1, maxIndex);
+        }
+
+        public int NormalizeSlotIndex(int index, int maxIndex)
+        {
+            return index > maxIndex ? 0 : (index < 0 ? maxIndex : index);
+        }
```

Comment:
Can we set some of these `helper` methods to be private inside the class?

Reply:
Just `GetNextIndex`